lib/AST/Printer/Printer.h

```diff
@@ -85,13 +100,21 @@ class PrintedTokenRangeImpl {
 
   // All allocated token contexts live here. The `TokenImpl::context_index` of
   // a `PrintedTokenImpl` points into `contexts`.
+  //
+  // NOTE(pag): The first context in this list holds a raw data pointer to the
+  //            `ASTImpl` containing `contexts` (if this is derived from an AST).
+  //            This is so that we can get the `Decl`s, `Stmt`s, and `Type`s
+  //            referenced by a token context. This is safe because a
+  //            `TokenContext` has a shared pointer to a vector of contexts
+  //            (this vector), where that shared pointer aliases the `ASTImpl`s
+  //            lifetime.
   std::vector<TokenContextImpl> contexts;
```

Comment:
Does that mean the first one is a dummy or is it still used the same way as the rest?
If it is a dummy wrap this in a helper struct that has it outside of the list.
If it is not I would still do wrap it and disallow erase operations to forbid accidental removal of the first one.